examples/pigweed-app/nrfconnect/README.md

```diff
@@ -0,0 +1,208 @@
+# CHIP nRF Connect nRF52840 Pigweed Example Application
+
+An example application showing the use
+[CHIP](https://github.com/project-chip/connectedhomeip) on the Nordic nRF52840.
+
+<hr>
+
+-   [CHIP nRF52840 Pigweed Example Application](#chip-nrf52840-lock-example-application)
+    -   [Introduction](#introduction)
+    -   [Building](#building)
+        -   [Using Docker container](#using-docker-container)
+        -   [Using Native shell](#using-native-shell)
+        -   [Supported nRF Connect SDK versions](#supported-nrf-connect-sdk-versions)
+    -   [Configuring the example](#configuring-the-example)
+    -   [Flashing and debugging](#flashing-and-debugging)
+    -   [Currently implemented features](#currently-implemented-features)
+
+<hr>
+
+<a name="intro"></a>
+
+## Introduction
+
+![nrf52840 DK](../../platform/nrf528xx/doc/images/nrf52840-dk.jpg)
+
+The nRF52840 Pigweed example app exercises functionalities in
+third_party/pigweed, to see what is needed for integrating Pigweed to CHIP, as
+well as a precursor to CHIP functionalities like on-device testing.
+
+The example makes use of the CMake build system to generate the ninja build
+script. The build system takes care of invoking the CHIP library build with all
+necessary flags exported from the Zephyr environment.
+
+<a name="building"></a>
+
+## Building
+
+### Using Docker container
+
+> **Important**:
+>
+> Due to
+> [certain limitations of Docker for MacOS](https://docs.docker.com/docker-for-mac/faqs/#can-i-pass-through-a-usb-device-to-a-container)
+> it is impossible to use the Docker container to communicate with a USB device
+> such as nRF 52840 DK. Therefore, MacOS users are advised to follow the
+> [Using Native shell](#using-native-shell) instruction.
+
+The easiest way to get started with the example is to use nRF Connect SDK Docker
+image for CHIP applications. Run the following commands to start a Docker
+container:
+
+        $ mkdir ~/nrfconnect
+        $ mkdir ~/connectedhomeip
+        $ docker pull nordicsemi/nrfconnect-chip
+        $ docker run --rm -it -v ~/nrfconnect:/var/ncs -v ~/connectedhomeip:/var/chip -v /dev/bus/usb:/dev/bus/usb --device-cgroup-rule "c 189:* rmw" nordicsemi/nrfconnect-chip
```

Comment:
This doesn't seem to work:

```
% docker run --rm -it -v ~/nrfconnect:/var/ncs -v ~/connectedhomeip:/var/chip -v /dev/bus/usb:/dev/bus/usb --device-cgroup-rule "c 189:* rmw" nordicsemi/nrfconnect-chip
Welcome to development environment for nRF Connect SDK and CHIP applications

- /var/ncs is empty

  to enable build of nRF Connect SDK applications run "setup" command which will fetch 
  required nRF Connect SDK sources

build@4014282afacf:/$ setup
/var/ncs repository is empty. Do you wish to check out nRF Connect SDK sources [master]? [Y/N] y
=== Initializing in /var/ncs
--- Cloning manifest repository from https://github.com/nrfconnect/sdk-nrf, rev. master
fatal: cannot mkdir /var/ncs/.west/manifest-tmp: Permission denied
FATAL ERROR: command exited with status 128: git init /var/ncs/.west/manifest-tmp

```

It probably works only if your uid outside docker is 1000.

Reply:
Can we document building without docker?

Reply:
I think VSCode container also assumes that your uid is 1000, right? It's kind of tricky to run a Docker container using the caller's uid since the Docker image must contain appropriate entries in `/etc/passwd`... The instruction for building without Docker is right below, but Docker is very helpful for people who don't want to spend time for installing nRF Connect dependencies. I may think about some workaround for your case - perhaps running the image as root and switching to the original user id in the entrypoint script would make it work for any user.

Reply:
I fixed that problem and updated the other nrfconnect READMEs: #3629. @kkasperczyk-no, could you apply the change to the pigweed-app as well?

Reply:
Sure, done.

Reply:
VSCode container does not assume your id is 1000.

Reply:
Hmm. .devcontainer/Dockerfile creates a user with the following data:
```
ARG USERNAME=vscode
ARG USER_UID=1000
ARG USER_GID=$USER_UID
ENV LANG en_US.utf8
```
and then:
``` "remoteUser": "vscode",``` in devcontainer.json. But I haven't tested how it behaves when started as a different user.

Reply:
`ARG` meaning arguments not constants.

Reply:
Ah, so VSCode container rebuilds the image for each user. I get it now.